app-backend/database/src/main/scala/com/azavea/rf/database/tables/ScenesToProjects.scala

```diff
@@ -175,4 +176,25 @@ object ScenesToProjects extends TableQuery(tag => new ScenesToProjects(tag)) wit
         .update(true)
     )
   }
+
+  /** All ingest locations for Scenes in a given Project.
+    * If even one of the Scenes doesn't have an `ingestLocation`, we fail
+    * the whole operation.
+    */
+  def allSceneIngestLocs(projectId: UUID)(implicit database: DB): Future[Option[List[(UUID, String)]]] = {
+    val sceneIds: Future[Seq[UUID]] = database.db.run {
+      ScenesToProjects
+        .filter(_.projectId === projectId)
+        .sortBy(_.sceneOrder.asc.nullsLast)
+        .map(_.sceneId)
+        .result
+    }
+
+    val scenes: Seq[UUID] => Future[Option[List[(UUID, String)]]] = { ids: Seq[UUID] =>
+      database.db.run(Scenes.filter(_.id inSet ids).map(s => s.ingestLocation.map((s.id, _))).result)
+        .map(_.toList.sequence)
+    }
```

Comment:
I didn't use `MosaicDefinition` at all for this, nor did I do a join. Just manual lookups with `inSet`.